libdrgn/program.h

```diff
@@ -64,9 +64,20 @@ struct vmcoreinfo {
 	bool pgtable_l5_enabled;
 };
 
+struct drgn_thread {
+	struct drgn_program *prog;
+	uint32_t tid;
+	union {
+		struct nstring prstatus;
+		struct drgn_object thread;
```

Comment:
I'd prefer the name `object` instead of `thread`: this is the object corresponding to this thread, not the thread itself.